core-war/src/main/java/org/silverpeas/web/look/AjaxServletLookV5.java

```diff
@@ -785,17 +785,21 @@ protected boolean isLoadingContentNeeded(UserMenuDisplay userMenuDisplayMode, St
   }
 
   /**
-   * a Space is visible if at least one of its items is visible for the currentUser
-   * @param userId
-   * @param spaceId
-   * @param helper
+   * a Space is visible if this space isn't removed and at least one of its items is visible for the
+   * current user
+   * @param userId the unique identifier of the user
+   * @param space the space
+   * @param helper a Look helper to access some peculiar settings regarding the accessibility.
    * @return true or false
    */
-  protected boolean isSpaceVisible(String userId, String spaceId, LookHelper helper) {
+  protected boolean isSpaceVisible(String userId, SpaceInstLight space, LookHelper helper) {
+    if (space.isRemoved()) {
```

Comment:
No. If a parent space is removed, all of its subspaces are also removed and their status is correctly valued at 'R' (for removed). In fact, the true pb is the status of the space wasn't fetched in some circumstances when getting a SpaceInstLight or a SpaceInst entity from the database. It was one fix I did. So, now, by just checking the status of a space, whatever it was explicitly removed or one of its parent, we can know if it is removed.

In another side, what it is true is the application instances aren't marked as removed when the space to which they belong is removed. So, for such situations, the use of the  `RemovedSpaceAndComponentInstanceChecker` class can be useful  until this issue with the component instances' status is fixed.

Nevertheless, I you prefer `RemovedSpaceAndComponentInstanceChecker` to be used here instead of `SpaceInstLight#isRemoved()`, I can do it. For me, here this class adds no value added and, more, adds useless little complexity (as it creates unused caches and it browses the parent tree of the concerned space to check for each node their status).

Reply:
I disagree with the behavior you described.

"Removed" information of a space (or the one of a component instance) is a flag indicating that the space (or component instance) which has been explicitly sent to administration basket.
This is this flag information that is used to display to administrators spaces and component instances to restore or to delete permanently.

When sent to basket, title of the space (or of the component instance) is updated, and "removed" flag is set. But nothing is updated about child data.

This mechanism worked successfully to display to users right spaces and right component instances he could access, until an issue was identified about the display of publication aliases that should have not (because main location was linked to a removed component instance). In order to resolve this alias problem, according to removed flag management, it has been chosen to implement RemovedSpaceAndComponentInstanceChecker which is able to compute logically if a space or a component instance is removed (taking care about potential huge data context).

But anyway, as I was not able to understand how it could work having in mind the mechanism, I tried your PR and checked that the fixing was indeed doing the stuff!
After a step by step debugging, I then realized that the 'isRemoved' method of a space was called into a forEach walking through a space tree...

So I finally agree with the fact that RemovedSpaceAndComponentInstanceChecker must not be used here!

Reply:
> If a parent space is removed, all of its subspaces are also removed and their status is correctly valued at 'R' (for removed).

> I disagree with the behavior you described. [...] But nothing is updated about child data.

Check the data in the table `st_spaces` and you will see the subspaces of a just removed parent space have also their status set to 'R'. The only child data that have their status unchanged are the component instances in the removed space and subspaces (as I wrote).